python/tvm/autotvm/record.py

```diff
@@ -246,7 +250,12 @@ def pick_best(in_file, out_file):
     out_file: str or file
         The filename of output
     """
-    best_context = ApplyHistoryBest(load_from_file(in_file))
+    context = load_from_file(in_file)
+    if os.path.isfile(out_file):
+        out_context = load_from_file(out_file)
+        context = itertools.chain(context, out_context)
```

Comment:
for a border condition, when the existing out_file is a old file and coming from a different hardware tune,  if we chain here and choose best later, is that may cause a hidden logic issue?

Reply:
It wont cause any issues but the resulting file will have schedules for both the old hardware and the new hardware. When using that file as a history, only the relevant schedules to the current build will be used. I argue this is desirable behavior as it prevents the old hardware jobs from being totally lost.